File: src/main/java/fr/marstech/mtlinkspray/entity/MtLinkSprayCollectionItem.java
package fr.marstech.mtlinkspray.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "mt-link-spray-collection")
public class MtLinkSprayCollectionItem {

    @Id
    private String id;

    private String description;

    private LocalDateTime creationDate;
}

File: src/main/kotlin/fr/marstech/mtlinkspray/entity/MtLinkSprayCollectionItem.kt
package fr.marstech.mtlinkspray.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime
import java.util.*

@Document(collection = "mt-link-spray-collection")
data class MtLinkSprayCollectionItem(
    @Id override val id: String = UUID.randomUUID().toString(),
    override val creationDate: LocalDateTime = LocalDateTime.now(),
    override val expiresAt: LocalDateTime? = null,
    override var isEnabled: Boolean = true,
    override var description: String? = null,
    override var metadata: MutableMap<String, String> = mutableMapOf(),
    override var author: HistoryItem,
    override var historyItems: MutableList<HistoryItem> = mutableListOf(),
) : StandardEntity
